### PR TITLE
[kube-prometheus-stack]: Allow kubeVersion to be overridden due to helm V3 rendering issue

### PR DESCRIPTION
**Pull request by viniciusps**
#### What this PR does / why we need it:

We are currently using ArgoCD to deploy our prometheus stack.

Recently we've had https://github.com/prometheus-community/helm-charts/commit/2c65605 merged in. (that's great)
However, It seems like Helm v3 does not support `.Capabilities.KubeVersion just yet.`  See: https://github.com/helm/helm/pull/9040.

This is the ArgoCD issue: https://github.com/argoproj/argo-cd/issues/5833.

So I am proposing an option to override the KubeVersion so that the ingress creation works.

Currently :
`{{- if and (.Capabilities.APIVersions.Has "networking.k8s.io/v1") (semverCompare ">= 1.19.x" .Capabilities.KubeVersion.Version) -}}`  defaults to v1.20 in our environment which does not match our current k8s version.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [ ] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ ] Chart Version bumped
- [ ] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/_helpers.tpl [hunk 1]
```diff
@@ -96,9 +96,18 @@ Allow the release namespace to be overridden for multi-namespace deployments in
...
+    {{- .Values.kubeVersionOverride -}}
+  {{- else -}}
+    {{- print .Capabilities.KubeVersion.Version -}}
+  {{- end -}}
+{{- end -}}
+
 {{/* Get Ingress API Version */}}
 {{- define "kube-prometheus-stack.ingress.apiVersion" -}}
-  {{- if and (.Capabilities.APIVersions.Has "networking.k8s.io/v1") (semverCompare ">= 1.19.x" .Capabilities.KubeVersion.Version) -}}
+{{- if and (.Capabilities.APIVersions.Has "networking.k8s.io/v1") (semverCompare ">= 1.19.x" (include "kube-prometheus-stack.ingress.kubeVersion" .)) -}}
```
Comment (JoshVee) [inline charts/kube-prometheus-stack/templates/_helpers.tpl:110]:
```suggestion
  {{- if and (.Capabilities.APIVersions.Has "networking.k8s.io/v1") (semverCompare ">= 1.19.x" (include "kube-prometheus-stack.ingress.kubeVersion" .)) -}}
```

**Review by JoshVee — APPROVED** [review]

**Review by JoshVee — APPROVED** [review]

File: charts/kube-prometheus-stack/templates/_helpers.tpl [hunk 2]
```diff
@@ -96,9 +96,18 @@ Allow the release namespace to be overridden for multi-namespace deployments in
   {{- end -}}
 {{- end -}}
 
+{{/* Allow KubeVersion to be overridden. */}}
+{{- define "kube-prometheus-stack.ingress.kubeVersion" -}}
+  {{- if .Values.kubeVersionOverride -}}
+    {{- .Values.kubeVersionOverride -}}
+  {{- else -}}
+    {{- print .Capabilities.KubeVersion.Version -}}
+  {{- end -}}
```
Comment (JoshVee) [inline charts/kube-prometheus-stack/templates/_helpers.tpl:105]:
```suggestion
  {{- print  .Values.kubeVersionOverride | default .Capabilities.KubeVersion.Version -}}
```

Comment (DirtyCajunRice) [reply charts/kube-prometheus-stack/templates/_helpers.tpl:105]: @JoshVee that is further simplified as we established earlier that all templates come back as strings
```
{{- default .Capabilities.KubeVersion.Version .Values.kubeVersionOverride -}}
```

Comment (viniciusps) [reply charts/kube-prometheus-stack/templates/_helpers.tpl]: Sounds good. Thanks.

Thanks for reviewing it. Would be great to have it merged if possible.

**Review by bismarck — APPROVED** [review]
lgtm

**Pull request merged**